core/src/models/AppRefreshQuery.ts

```diff
@@ -73,21 +73,38 @@ export class AppRefreshQuery extends LoggedModel<AppRefreshQuery> {
       appId: this.appId,
       refreshQuery: this.refreshQuery,
       value: this.value,
+      locked: this.locked,
+      state: this.state,
       lastChangedAt: APIData.formatDate(this.lastChangedAt),
       lastConfirmedAt: APIData.formatDate(this.lastConfirmedAt),
       createdAt: APIData.formatDate(this.createdAt),
       updatedAt: APIData.formatDate(this.updatedAt),
     };
   }
 
+  async test(refreshQuery?: string) {
+    return AppRefreshQueryOps.test(this, refreshQuery);
+  }
+
   // --- Class Methods --- //
 
   static async findById(id: string) {
     const instance = await this.scope(null).findOne({ where: { id } });
     if (!instance) throw new Error(`cannot find AppRefreshQuery ${id}`);
     return instance;
   }
-
+  @BeforeCreate
+  static async ensureOnePerApp(instance: AppRefreshQuery) {
+    const existingCount = await AppRefreshQuery.scope(null).count({
+      where: {
+        appId: instance.appId,
+      },
+    });
+
+    if (existingCount > 0) {
+      throw new Error(`app ${instance.appId} already has an app refresh query`);
+    }
+  }
```

Comment:
Because you added a unique index in `core/src/migrations/000090-appRefreshQueryUniqueAppId.ts`, you don't need this check.  The database will enforce that there's only one AppRefreshQuery per AppId.  This can be removed.  The specific error message will be different, but the logic is the same.